app/questionnaire/router.py

```diff
@@ -66,44 +66,28 @@ def can_access_hub(self):
     def routing_path(self, section_id, list_item_id=None):
         return self._path_finder.routing_path(section_id, list_item_id)
 
-    def get_next_location_url(self, location, routing_path):
+    def get_next_location_url(self, location, routing_path, return_to_summary=False):
         """
         Get the first incomplete block in section/survey if trying to access the section/survey end,
         and the section/survey is incomplete or gets the next default location if the above is false.
         """
-        current_block_type = self._schema.get_block(location.block_id)["type"]
         last_block_id = routing_path[-1]
-        last_block_type = self._schema.get_block(last_block_id)["type"]
         hub_enabled = self._schema.is_hub_enabled()
+        has_section_summary = self._schema.is_summary_in_section(location.section_id)
 
         if (
-            hub_enabled
-            and location.block_id == last_block_id
-            and self._progress_store.is_section_complete(
-                location.section_id, location.list_item_id
-            )
+            return_to_summary or last_block_id == location.block_id
+        ) and self._progress_store.is_section_complete(
+            location.section_id, location.list_item_id
         ):
-            return url_for(".get_questionnaire")
 
-        # If the section is complete and contains a SectionSummary, return the SectionSummary location
-        if (
-            last_block_type == "SectionSummary"
-            and current_block_type != last_block_type
-            and self._progress_store.is_section_complete(
-                location.section_id, location.list_item_id
-            )
-        ):
-            return url_for(
-                "questionnaire.block",
-                block_id=last_block_id,
-                list_name=routing_path.list_name,
-                list_item_id=routing_path.list_item_id,
-            )
+            if has_section_summary:
+                return self.get_section_summary_url(location)
+            if hub_enabled:
+                return url_for(".get_questionnaire")
```

Comment:
We should be consistent with these, one is a method of router, the other just uses `url_for`. The questionnare url is commonly used so it makes sense for them to all live in the same place.

Reply:
Not sure I understand, both are using url_for in router.py, I just extracted the top one into its own static method. I will re-combine to make it clearer

Reply:
As in, we now have a method which just wrap `url_for` for section summary url. Whereas, get_questionnaire is fetched directly. I think it would be clean to have a wrapper method of any use of url_for since it is commonly used.

Reply:
I have folded this back in for now, but Router does this in 6 places across the class and has 3 different flavours, for section, for block and for questionnaire, so we would have to unpack args in some way if we wanted to do this. This to me sounds like a class refactor, as there are a number of things in router that need a second look.